default/python/AI/InvasionAI.py

```diff
@@ -436,7 +436,8 @@ def evaluate_invasion_planet(planet_id, secure_fleet_missions, verbose=True):
                           2*planet.currentMeterValue(fo.meterType.targetResearch))
 
     # devalue invasions that would require too much military force
-    threat_factor = min(1, 0.2*MilitaryAI.get_tot_mil_rating()/(sys_total_threat+0.001))**2
+    threat_factor = min(1, MilitaryAI.get_preferred_max_military_portion_for_single_battle() *
+                        MilitaryAI.get_concentrated_tot_mil_rating()/(sys_total_threat+0.001))**2
```

Comment:
This formula is kind of hard to read with the long function calls. Could be worthwhile to add variables for better readability.